cardano-db-sync/src/Cardano/DbSync/Era/Universal/Insert/GovAction.hs

```diff
@@ -124,17 +128,39 @@ insertGovActionProposal cache blkId txId govExpiresAt (index, pp) = do
           , DB.treasuryWithdrawalAmount = Generic.coinToDbLovelace coin
           }
 
-    insertNewCommittee gaId removed added q = do
-      void . DB.insertNewCommittee $
-        DB.NewCommittee
-          { DB.newCommitteeGovActionProposalId = gaId
-          , DB.newCommitteeQuorumNumerator = fromIntegral $ numerator r
-          , DB.newCommitteeQuorumDenominator = fromIntegral $ denominator r
-          , DB.newCommitteeDeletedMembers = textShow removed
-          , DB.newCommitteeAddedMembers = textShow added
+    insertNewCommittee ::
+      DB.GovActionProposalId ->
+      Set (Ledger.Credential 'ColdCommitteeRole StandardCrypto) ->
+      Map (Ledger.Credential 'ColdCommitteeRole StandardCrypto) EpochNo ->
+      UnitInterval ->
+      ReaderT SqlBackend m ()
+    insertNewCommittee gapId removed added q = do
+      insertNewCommitteeInfo gapId q
+      insertMembers gapId removed added q
+
+    insertNewCommitteeInfo gapId q =
+      void . DB.insertNewCommitteeInfo $
+        DB.NewCommitteeInfo
+          { DB.newCommitteeInfoGovActionProposalId = gapId
+          , DB.newCommitteeInfoQuorumNumerator = fromIntegral $ numerator r
+          , DB.newCommitteeInfoQuorumDenominator = fromIntegral $ denominator r
           }
       where
         r = unboundRational q -- TODO work directly with Ratio Word64. This is not currently supported in ledger
+    insertMembers gapId removed added q = do
+      whenJust mmCommittee $ \mCommittee -> do
+        -- Nothing means we're not in Conway so it can't happen.
+        let committee = updatedCommittee removed added q mCommittee
+        mapM_ (insertNewMember gapId) (Map.toList $ committeeMembers committee)
+
+    insertNewMember gapId (cred, e) = do
+      chId <- insertCommitteeHash cred
```

Comment:
Would there ever be a situation where inserting the Committee hash would fail and we need to handle it?

Reply:
Internally it uses `insertCheckUnique`, so if the entry already exists, it will return the existing id.

Reply:
perfect thanks